packages/logger/README.md

```diff
@@ -0,0 +1,39 @@
+# @firebase/logger
+
+This package serves as the base of all logging in the JS SDK. Any logging that
+is intended to be for firebase end developers, should go through this module.
+
+## Basic Usage
+
+Firebase components should import the `Logger` class and instantiate a new
+instance by passing a component name (e.g. `@firebase/<COMPONENT>`) to the
+constructor.
+
+_e.g._
+
+```typescript
+import { Logger } from '@firebase/logger';
+
+const logClient = new Logger(`@firebase/<COMPONENT>`);
+```
+
+Each `Logger` instance supports 5 log functions each to be used in a specific
+instance:
+
+- `debug`: Internal logs; use this to allow developers to send us their debug
+  logs for us to be able to diagnose an issue.
```

Comment:
I've left a comment in `logger.ts` to explain this and I am now intentionally routing `debug` logs to `console.log`

Reply:
If there's an action item for the user, wouldn't we want the log on by default?  INFO logs are off by default, so that would not be appropriate.

But my bigger problem is that "LOG" is meaningless since these are _all_ logs and I've never come across another logging system that has "LOG" as its own log level so nobody will know what this means or where it fits in the overall ordering. Even other JavaScript loggers (where `log` as a loglevel might almost make sense because of the `console.log` legacy) don't have it...

[winston](https://github.com/winstonjs/winston) [10k stars] - error, warn, info, verbose, debug
[bunyan](https://github.com/trentm/node-bunyan) [5k stars] - fatal, error, warn, info, debug, trace
[loglevel](https://github.com/pimterry/loglevel) [1k stars] - error, warn, info, debug, trace

So I appreciate that you've made up your own level and have strawmen examples to back it up, but I think it would be better to stick with well-known ones.

I think in practice:
1. 90% of customers will stick with the default (warn and error) all the time.
2. 10% of customers may enable debug logging because support asked them to or they found a StackOverflow / forum post that mentions it.
3. 0% of customers will care to distinguish between info/log/debug.
4. Only Firestore and RTDB use these logs and they only have error, warn, and debug. So there's currently no distinction to make anyway.

All that said, I don't really care. I think we should avoid LOG. If you think we need 5 log levels instead of 4 or 3, then please pick from the existing set of log levels that other systems use (notice, verbose, trace, etc.) for your 5th.  Keep in mind we could always add more log levels later if we need them.